src/lib/moment/constructor.js

```diff
@@ -54,19 +54,15 @@ export function copyConfig(to, from) {
     return to;
 }
 
-var updateInProgress = false;
-
-// Moment prototype object
+// Moment prototype object.
+//
+// Run hooks.updateOffset(new Moment(config)) if you're constructing from user
+// input, but be careful to detect and break out of loops in case the user's
+// version of hooks.updateOffset() decides to trigger the same code path.
+// (Or just use from-anything's createFromConfig(), which handles this for you.)
 export function Moment(config) {
     copyConfig(this, config);
     this._d = new Date(config._d != null ? config._d.getTime() : NaN);
-    // Prevent infinite loop in case updateOffset creates new moment
-    // objects.
-    if (updateInProgress === false) {
-        updateInProgress = true;
-        hooks.updateOffset(this);
-        updateInProgress = false;
-    }
```

Comment:
I moved this code to `createFromConfig`, which seems to cover all cases of creating Moments from non-Moment input.  In doing so, I'm assuming that it's safe to clone any Moment without running `updateOffset`.  That seems like a reasonably safe assumption, but I'd like to highlight it in case there's anything I'm not aware of.

When I kept the updateOffset call here, I had a subtle bug with the way moments got cloned when the `parseZone` algorithm calls the wrapped copy of utcOffset from our prototype.  It occurs to me now that I probably could've fixed that by (1) re-ordering operations within `parseZone`, and/or (2) using the unwrapped implementation behind `utcOffset` instead of simply calling the prototype's copy.  I'd be happy to try replacing this with one or both of those solutions if you're concerned about this particular edit -- but if this edit seems safe, I think it's probably faster and easier to reason about.

(Note that this edit avoids the messiness of trying to make an immutable updateOffset work inside the constructor:

```js
updateInProgress = true;
var updated = hooks.updateOffset(this);
copyConfig(this, updated);
this._d = updated._d;
updateInProgress = false;
```

Which is ugly and annoying.)